test/services/ProcessCsvServiceSpec.scala

```diff
@@ -0,0 +1,390 @@
+        .runWith(Sink.seq).map { fileCopied =>
+        testProcessCsvService.processRow(fileCopied.flatten.toList, "Other_Grants_V3.csv", dataValidator)
+      }
+
+      val result = Await.result(resultFuture, Duration.Inf)
+      result match {
+        case Right(validationErrorList) =>
+          validationErrorList.validationErrors.head.cell mustBe Cell("A", 0, "25  2015")
+          validationErrorList.validationErrors.head.errorId mustBe "001"
+          validationErrorList.validationErrors.head.errorMsg mustBe "ers.upload.error.date"
+          validationErrorList.validationErrors.head.ruleId mustBe "error.1"
+        case Left(ex) => throw ex
+      }
+    }
+
+    "process a row and return an empty list if there are no errors" in {
+      val source = convertToAkkaSource(new File(System.getProperty("user.dir") + "/test/resources/copy/CSOP_OptionsGranted_V3.csv"))
+      val dataValidator = new DataValidator(ConfigFactory.load.getConfig("ers-csop-granted-validation-config"))
+      val resultFuture = source
+        .runWith(Sink.seq).map { fileCopied =>
+        testProcessCsvService.processRow(fileCopied.flatten.toList, "CSOP_OptionsGranted_V3.csv", dataValidator)
+      }
+
+      val result = Await.result(resultFuture, Duration.Inf)
+      result match {
+        case Right(validationErrorList) => validationErrorList.validationErrors.isEmpty mustBe true
+        case Left(ex) => throw ex
+        }
+      }
+  }
+
+  "processFiles" should {
+
+    implicit val request: RequestWithOptionalEmpRef[AnyContent] = RequestWithOptionalEmpRef(FakeRequest(), None)
+    when(mockDataGenerator.getSheetCsv(any(), any())(any())).thenReturn(Right((ERSTemplatesInfo.ersSheets("CSOP_OptionsGranted_V3"), "csop")))
+    when(mockDataGenerator.identifyAndDefineSheetEither(any())(any(), any(), any())).thenReturn(Right("CSOP_OptionsGranted_V3"))
+    when(mockDataGenerator.setValidatorCsv(any())(any(), any(), any())).thenReturn(Right(new DataValidator(ConfigFactory.load.getConfig("ers-csop-granted-validation-config"))))
+    when(mockErsUtil.SCHEME_ERROR_COUNT_CACHE).thenReturn("10")
+    when(mockErsUtil.cache[Any](any(), any())(any(), any(), any(), any())).thenReturn(Future(CacheMap("1", Map("test" -> Json.obj("test" -> "test")))))
+
+    def returnStubSource(x: String, data: String): Source[HttpResponse, NotUsed] = {
+      Source.single(HttpResponse(entity = data))
+    }
+
+    "return true if all rows are valid" in {
+      val callback = UpscanCsvFilesCallbackList(List(UpscanCsvFilesCallback(UploadId("1"), UploadedSuccessfully("CSOP_OptionsGranted_V3.csv", "no", Some(1)))))
+      val data = "2015-09-23,250,123.12,12.1234,12.1234,no,yes,AB12345678,no\n2015-09-23,250,123.12,12.1234,12.1234,no,yes,AB12345678,no\n2015-09-23,250,123.12,12.1234,12.1234,no,yes,AB12345678,no"
+
+      val resultFuture = testProcessCsvService.processFiles(Some(callback), "csop", returnStubSource(_ , data))
+
+      val result = Await.result(resultFuture, Duration.Inf)
```

Comment:
Not something you have to change but there is a lowercase `'await'` that does this Await.Result that comes as part of UnitSpec.

Reply:
tbh UnitSpec is from hmrctest which is deprecated now - we didn't use it to make the move away from it in the future less painful